automation/prepare-and-start.sh

```diff
@@ -0,0 +1,33 @@
+#!/bin/bash
+
+set -e
+
+DOCKER_TIMEOUT=20 # Wait 20 seconds for docker to start
+
+# Create the normal user to be used as builder
+echo "[INFO] Creating and setting builder user $BUILDER_UID:$BUILDER_GID."
+groupadd docker
+groupadd -g $BUILDER_GID builder
+useradd -u $BUILDER_UID -g $BUILDER_GID -G docker builder
+
+# Start docker
+echo "[INFO] Starting docker."
+docker daemon 2> /dev/null &
+echo "[INFO] Waiting for docker to initialize..."
+STARTTIME=$(date +%s)
+ENDTIME=$(date +%s)
+while [ ! -S /var/run/docker.sock ]
+do
+    if [ $(($ENDTIME - $STARTTIME)) -le $DOCKER_TIMEOUT ]; then
+        sleep 1
+        ENDTIME=$(date +%s)
+    else
+        echo "[ERROR] Timeout while waiting for docker to come up."
+        exit 1
+    fi
+done
+echo "[INFO] Docker was initialized."
+
+# Start barys with all the arguments requested
+echo "[INFO] Running build as builder user..."
+sudo -H -u builder /yocto/resin-board/resin-yocto-scripts/build/barys $@
```

Comment:
After this finishes we should need to send a SIGTERM to docker and wait for it to exit before having the container exits so that docker can release the resources/loop devices it takes control of.